src/Parallel/CreateFromOptions.hpp

```diff
@@ -3,19 +3,31 @@
 
 #pragma once
 
+#include "Options/Auto.hpp"
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TaggedTuple.hpp"
 #include "Utilities/TypeTraits.hpp"
 
 namespace Parallel {
 namespace detail {
+template <typename T, typename Label>
+const std::optional<T>& get_option(const Options::Auto<T, Label>& option) {
+  return option;
+}
+
+template <typename T>
+const T& get_option(const T& option) {
+  return option;
+}
+
 template <typename Metavariables, typename Tag, typename... OptionTags,
           Requires<Tag::pass_metavariables> = nullptr>
 typename Tag::type create_initialization_item_from_options(
     const tuples::TaggedTuple<OptionTags...>& options) {
   return tuples::apply<typename Tag::template option_tags<Metavariables>>(
       [](const auto&... option) {
-        return Tag::template create_from_options<Metavariables>(option...);
+        return Tag::template create_from_options<Metavariables>(
+            get_option(option)...);
```

Comment:
I don't like special-casing `Auto` here.  It isn't explicitly converted in the more common option parsing locations, and this should be kept consistent with those.

It is unfortunate that C++ can't figure out to do the implicit conversion in some cases, but I think it is the job of the class requesting an `Auto` to be able to handle receiving one.

Reply:
Yeah this was very unfortunate that it couldn't figure it out. I'll remove this commit and try some other magic in the tags themselves.